### DISCUSSION TITLE
How to access the return value from HTTP command

### DISCUSSION THREAD

**Discussion by FlexMcMurphy** [arendst/Tasmota#21859] [Q&A] [answered]
Hello,

I am making a python script that issues a power off and power on command to a Tasmota smart plug.

I am trying to put some error handling in to capture if the Tasmota command completed successfully.

If I execute this command FROM A LINUX BASH TERMINAL to turn off the smartplug but mis-spell the Tasmota 'Power' command to try and simulate an error:
`curl -sSf http://192.168.1.189/cm?cmnd=PowerFUL%20off`

I get back json output: `{"Command":"Unknown"}` to the command line indicating an error but if I then do `#echo $?` I get `'0'` which I suppose just means that the curl command completed successfully.

To execute that curl command to turn off the smart plug using HTTP I use this code in my python script, with the Python subprocess module, again I mis-spelt the Tasmota command to try to simulate an error:

```
cmd = "curl -sSf http://192.168.1.189/cm?cmnd=PowerFUL%20off && echo success || echo fail"

output = Popen(cmd, shell=True, executable='/bin/bash', stdout=PIPE, stderr=PIPE, close_fds=False, text=True)

(strOut, strErr) = output.communicate()

print("Smart Plug Turn Off Output: " + "Err Message: " + strErr + " Out Message: " + strOut)
```

I get this output: `"Smart Plug Turn Off Output: Err Message:  Out Message: {"Command":"Unknown"}success"`

Which, I think, shows that the json output: `{"Command":"Unknown"}` does not get sent to stderr but only to stdout where it ends up getting concatenated to the stdout from the `echo` command.

Is there a way to capture the json feedback of running a Tasmota command via an HTTP url?
In my python script should I be using json to issue the Tasmota commands instead of the python subprocess module?

Cheers,

Flex

**Comment by barbudor** [comment, marked as answer]
First, why spaning a curl command instead of using [requests](https://fr.python-requests.org/en/latest/)?

Tasmota will always answer 200 OK to a command submitted through HTTP 
So you need to get the command exécution status by analysing the JSON return

**Reply by FlexMcMurphy** [reply]
@barbudor 
Merci beaucoup!

So here is my new python code using json instead of the python subprocess module.
It gets the power status of the smart plug after issuing the Tasmota power off command.
`print(response.text)` gives back `{'POWER': 'OFF'}` and that's all I need to know.

```
#!/usr/bin/python3

import requests

def restartZgemma():

    # Power OFF the Smart Plug (and hence the ZGemma)
    response = requests.get('http://192.168.1.189/cm?cmnd=Power%20off')
    parsed = response.json()

    print(parsed)
    print(response.status_code)
    print(response.text)

def main():

    restartZgemma()

# Python script starts here
if __name__ == '__main__':
    main()
```

Flex
